File: ProjectBuild/wwwroot/shared/client-side-models/buildModels.js
"use strict";
exports.__esModule = true;
var Dependency = /** @class */ (function () {
    function Dependency() {
    }
    return Dependency;
}());
exports.Dependency = Dependency;
var TextMessage = /** @class */ (function () {
    function TextMessage() {
    }
    return TextMessage;
}());
exports.TextMessage = TextMessage;
var CellCarrier = /** @class */ (function () {
    function CellCarrier() {
    }
    return CellCarrier;
}());
exports.CellCarrier = CellCarrier;
var AppSettings = /** @class */ (function () {
    function AppSettings() {
        this.debug = false;
        this.testing = false;
        this.connectionString = "";
        this.projectVersionNo = "";
        this.splashTime = 0;
        this.googleMapKey = "";
        this.smtpReply = "";
        this.smtpHost = "";
        this.smtpPort = 0;
        this.smtpUn = "";
        this.smtpPw = "";
        this.cellCarriers = "";
        this.aspNetCoreVersion = "";
    }
    return AppSettings;
}());
exports.AppSettings = AppSettings;
var AngularProject = /** @class */ (function () {
    function AngularProject() {
        this.visualProject = "";
        this.name = "";
        this.buildEnabled = false;
        this.pwaSupport = false;
        this.production = false;
        this.distFolder = "";
        this.angularModule = "";
        this.angularRoot = "";
        this.angularProjectDir = "";
        this.showPanel = false;
    }
    return AngularProject;
}());
exports.AngularProject = AngularProject;
var LaunchSettings = /** @class */ (function () {
    function LaunchSettings() {
    }
    return LaunchSettings;
}());
exports.LaunchSettings = LaunchSettings;
var DeveloperSettings = /** @class */ (function () {
    function DeveloperSettings() {
        this.machineName = "";
        this.buildHook = false;
        this.importHook = false;
        this.executeDist = false;
        this.serveApp = "";
        this.releaseApp = "";
        this.libraryExports = Array();
        this.angularProjects = Array();
    }
    return DeveloperSettings;
}());
exports.DeveloperSettings = DeveloperSettings;
var VisualProject = /** @class */ (function () {
    function VisualProject() {
        this.name = "";
        this.projectVersionNo = "";
        this.applicationUrl = "";
        this.workingDirectory = "";
        this.developerSettings = new DeveloperSettings();
        this.showPanel = false;
        this.showVersion = true;
    }
    return VisualProject;
}());
exports.VisualProject = VisualProject;
var BuildConfiguration = /** @class */ (function () {
    function BuildConfiguration() {
        this.machineName = "";
        this.visualProjects = Array();
        this.shared = Array();
    }
    return BuildConfiguration;
}());
exports.BuildConfiguration = BuildConfiguration;
var BuildResponse = /** @class */ (function () {
    function BuildResponse() {
        this.consoleWindow = "";
        this.versionNo = "";
    }
    return BuildResponse;
}());
exports.BuildResponse = BuildResponse;

File: BuildModels.cs
using System.Collections.Generic;

namespace Angular.Net.CLI.Models
{
    public class ApiVersions
    {
        public string typeScript { get; set; }
        public string nodeJs { get; set; }
        public string v8Engine { get; set; }
        public string angular { get; set; }
        public string rxJs { get; set; }
        public string lodash { get; set; }
        public string moment { get; set; }
        public string fileSaver { get; set; }
        public string coreJs { get; set; }
        public string zoneJs { get; set; }
        public string googleMaps { get; set; }
        public string ngxtoastr { get; set; }
    }

    public class AppSettings
    {
        public string launchPath { get; set; }
        public bool debug { get; set; }
        public bool testing { get; set; }
        public bool onlineStatus { get; set; }
        public string connectionString { get; set; }
        public string buildVersion { get; set; }
        public int splashTime { get; set; }
        public string googleMapKey { get; set; }
        public string smtpReply { get; set; }
        public string smtpHost { get; set; }
        public int smtpPort { get; set; }
        public string smtpUn { get; set; }
        public string smtpPw { get; set; }
        public string cellCarriers { get; set; }
        public string aspNetCoreVersion { get; set; }
        public ApiVersions apiVersions { get; set; }
    }

    public class CellCarrier
    {
        public string name { get; set; }
        public string smsProfile { get; set; }
    }

    public class AngularProject
    {
        public string visualProject { get; set; }
        public string name { get; set; }
        public bool buildEnabled { get; set; }
        public bool pwaSupport { get; set; }
        public bool production { get; set; }
        public string distFolder { get; set; }
        public string angularModule { get; set; }
        public string angularRoot { get; set; }
        public string angularProjectDir { get; set; }
        public bool showPanel { get; set; }
    }

    public class BuildResponse
    {
        public string payloadType { get; set; }
        public string outputStream { get; set; }
        public string consoleWindow { get; set; }
        public string versionNo { get; set; }
    }

    public class DeveloperSettings
    {
        public string machineName { get; set; }
        public bool buildHook { get; set; }
        public bool importHook { get; set; }
        public bool executeDist { get; set; }
        public string serveApp { get; set; }
        public string releaseApp { get; set; }
        public List<string> libraryExports { get; set; }
        public List<AngularProject> angularProjects { get; set; }
    }

    public class VisualProject
    {
        public string name { get; set; }
        public string applicationUrl { get; set; }
        public string workingDirectory { get; set; }
        public DeveloperSettings developerSettings { get; set; }
        public bool showPanel { get; set; }
        public bool showVersion { get; set; }
    }

    public class BuildConfiguration
    {
        public string machineName { get; set; }
        public List<VisualProject> visualProjects { get; set; }
        public List<string> shared { get; set; }
    }


}
